react/react1/week3/reactweek3/src/TodoList.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from "react";
+import TodoItem from "./TodoItem";
+// import { DataFetching } from "./DataFetching";
+
+export function TodoList() {
+  const [useTodo, setUseTodo] = useState([]);
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const Api =
+    "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+  useEffect(() => {
+    if (setUseTodo) {
+      fetch(Api)
+        .then((res) => res.json())
+        .then((data) => {
+          const fetchMap = data.map((data) => ({
+            id: data.id,
+            description: data.description,
+            deadline: data.deadline,
+          }));
+
+          setUseTodo((prev) => [...prev, ...fetchMap]);
+        });
+    }
+  }, [setUseTodo]);
+
+  // for add State
+  function addTodo() {
+    let addID;
+    if (useTodo.length === 0) {
+      addID = 1;
+    } else {
+      addID = useTodo[useTodo.length - 1].id + 1;
+    }
```

Comment:
This is fine. However you could make it a bit more concise by using the ternary operator.

`const id = useTodo.length === 0 ? 1 : useTodo[useTodo.length - 1].id + 1;`

Reply:
done